astropy/table/table.py

```diff
@@ -1778,89 +1778,58 @@ def _make_index_row_display_table(self, index_row_name):
-            will be retrieved from the configuration item
-            ``astropy.table.default_notebook_table_class``. Note that these
-            table classes may make use of bootstrap, as this is loaded with the
-            notebook.  See `this page <https://getbootstrap.com/css/#tables>`_
-            for the list of classes.
-        css : str
-            A valid CSS string declaring the formatting for the table. Defaults
-            to ``astropy.table.jsviewer.DEFAULT_CSS_NB``.
-        display_length : int, optional
-            Number or rows to show. Defaults to 50.
-        show_row_index : str or False
-            If this does not evaluate to False, a column with the given name
-            will be added to the version of the table that gets displayed.
-            This new column shows the index of the row in the table itself,
-            even when the displayed table is re-sorted by another column. Note
-            that if a column with this name already exists, this option will be
-            ignored. Defaults to "idx".
+        backend : {"ipydatagrid", "classic"}
+            Backend to use for rendering (default="ipydatagrid"). The "classic" backend
+            is deprecated since v6.1.
+
+        **kwargs : dict, optional
+            Keyword arguments as accepted by desired backend. See `astropy.table.notebook_backends`
+            for the available backends and their respective keyword arguments.
+
+        Raises
+        ------
+        NotImplementedError
+            Requested backend is not supported.
+
+        See Also
+        --------
+        astropy.table.notebook_backends
 
-        Notes
-        -----
-        Currently, unlike `show_in_browser` (with ``jsviewer=True``), this
-        method needs to access online javascript code repositories.  This is due
-        to modern browsers' limitations on accessing local files.  Hence, if you
-        call this method while offline (and don't have a cached version of
-        jquery and jquery.dataTables), you will not get the jsviewer features.
         """
-        from IPython.display import HTML
+        if backend == "ipydatagrid":
+            from astropy.table.notebook_backends import ipydatagrid
 
-        from .jsviewer import JSViewer
+            func = ipydatagrid
 
-        if tableid is None:
-            tableid = f"table{id(self)}-{np.random.randint(1, 1e6)}"
+        elif backend == "classic":
+            from astropy.table.notebook_backends import classic
+
+            # NOTE: The leading whitespace in warning lines is for backward compatibility.
+            warnings.warn(
+                """show_in_notebook() is deprecated as of 6.1 and to create
+         interactive tables it is recommended to use dedicated tools like:
+         - https://github.com/bloomberg/ipydatagrid
+         - https://docs.bokeh.org/en/latest/docs/user_guide/interaction/widgets.html#datatable
+         - https://dash.plotly.com/datatable""",
```

Comment:
I am not sure about this change either. I kept the message backward compatible on purpose. I don't see a strong reason to change a deprecation text that has been released into v6.1.

Reply:
The text states directly that the `show_in_notebook()` method is deprecated, and that is no longer the case. Only using the `classic` backend is now deprecated, while the default `ipydatagrid` backend is fully supported and endorsed.

The recommendation of alternate 3rd-party packages was driven by having no supported solution built-in to astropy. Since that is no longer the case with this PR, we should remove that text, right?

Reply:
Ah right...